core/templates/services/i18n-language-code.service.ts

```diff
@@ -53,6 +55,28 @@ export class I18nLanguageCodeService {
         this.getCurrentI18nLanguageCode()) !== -1);
   }
 
+  // Remove this method when translation service is extended.
+  getClassroomTranslationKey(classroomName: string): string {
+    return `I18N_CLASSROOM_${classroomName.toUpperCase()}_TITLE`;
+  }
+
+  // Remove this method when translation service is extended.
+  getTopicTranslationKey(topicId: string, isDescription: boolean): string {
+    return (
+      isDescription ?
+      `I18N_TOPIC_${topicId}_DESCRIPTION` :
+      `I18N_TOPIC_${topicId}_TITLE`
+    );
+  }
+
+  // Remove this method when translation service is extended.
+  hasTranslations(translationKey: string): boolean {
```

Comment:
I think a docstring is needed here. I'm not sure what this method is trying to evaluate, conceptually.

Reply:
Have added docstring explaining what it does, so as we need to add translation keys to the constants file if we want to show it's translation this function basically checks if the key passed to it exists in the constants file or not, if it does it would mean the key has a translation or else it does not have one

Reply:
Then call it what it is -- e.g. doesTranslationKeyExistInConstantsFile or doesTranslationKeyHaveHackyTranslations or similar. (I'm not sure these names are quite right, but hopefully they give you some ideas -- the name should represent very clearly what "being in the constants file" actually means.)

Reply:
Made the change